src/endpoints/esdt/esdt.address.service.ts

```diff
@@ -217,14 +216,14 @@ export class EsdtAddressService {
     if (filter.identifiers && filter.identifiers.length === 1) {
       const identifier = filter.identifiers[0];
 
-      if (identifier.split('-').length === 2) {
+      if (TokenUtils.isSovereignIdentifier(identifier)) {
+        collection = identifier.split('-').slice(0, 3).join('-');
+        nonceHex = identifier.split('-')[3];
+      } else {
         collection = identifier.split('-').slice(0, 2).join('-');
         nonceHex = identifier.split('-')[2];
       }
```

Comment:
you can avoid identifier.split('-') by adding into a const like:

const parts = identifier.split('-')

Reply:
fixed